File: backup_before_industrialisation_20250626/backend/components/metiers/banque_finance/utils/audit/fallback/fallback.py
# fallback.py
# Fallback d'audit Python pour Banque_Finance – exemple clé en main
from datetime import datetime

fallback_logs = []


def audit_fallback(action, details=None):
    """
    Fallback minimal : log d'audit en mémoire si la persistance échoue
    """
    if details is None:
        details = {}
    log = {
        "timestamp": datetime.utcnow().isoformat() + "Z",
        "action": action,
        **details,
    }
    fallback_logs.append(log)
    return log

File: backend/components/metiers/voice/utils/audit/fallback/fallback.js
// fallback.js
// Fallback d'audit JS pour voice – exemple clé en main

/**
 * Fallback minimal : log d'audit en mémoire si la persistance échoue
 * @param {string} action
 * @param {object} details
 * @returns {object}
 */
const fallbackLogs = [];
function auditFallback(action, details = {}) {
  const log = {
    timestamp: new Date().toISOString(),
    action,
    ...details
  };
  fallbackLogs.push(log);
  return log;
}

module.exports = { auditFallback, fallbackLogs };
